Documentation for air-gapped support (#69)
project-koku/koku-metrics-operator: docs/csv-description.md

#### Limitations (Potential for metrics data loss)
* A source **must** exist in cloud.redhat.com for an uploaded payload to be processed by cost management. The operator sends the payload to the Red Hat Insights Ingress service which usually returns successfully, but the operator does not currently confirm with cost management that the payload was processed. After Ingress accepts the uploaded payload, the payload is removed from the operator and is gone forever. If the data within the payload is not processed, a gap will be introduced in the usage metrics.

**Note** The following limitations are specific to operators configured to run in a restricted network: 
* The `koku-metrics-operator` will not be able to generate new reports if the PVC storage is filled. If this occurs, the reports must be manually deleted from the PVC so that the operator can function as normal. 
* The default report retention is 30 reports (about one week's worth of data). The reports must be manually downloaded and uploaded to cloud.redhat.com every week, or they will be deleted and the data will be lost. 

> #### Storage configuration pre-requisite

Reviewer: ```suggestion
#### Storage configuration prerequisite
```

The operator will attempt to create and use the following PVC when installed:
  ```
    volume_claim_template:
      apiVersion: v1
      kind: PersistentVolumeClaim
      metadata:
        name: koku-metrics-operator-data
      spec:
        accessModes:
          - ReadWriteOnce
        resources:
          requests:
            storage: 10Gi
  ```
If a different PVC should be utilized, a valid PVC should be specified in the KokuMetricsConfig CR as described in the following section. The PVC to be used may exist already, or the operator will attempt to create it.

To use the default specification, the follow assumptions must be met:
1. A default StorageClass is defined.

> 2. Dynamic provisioning for that default StorageClass is enabled.

Reviewer: ```suggestion
2. Dynamic provisioning for that default StorageClass is enabled.

```

**Note** The following limitations are specific to operators configured to run in a restricted network: 
* The `koku-metrics-operator` will not be able to generate new reports if the PVC storage is filled. If this occurs, the reports must be manually deleted from the PVC so that the operator can function as normal. 
* The default report retention is 30 reports (about one week's worth of data). The reports must be manually downloaded and uploaded to cloud.redhat.com every week, or they will be deleted and the data will be lost. 

#### Storage configuration pre-requisite
The operator will attempt to create and use the following PVC when installed:
  ```
    volume_claim_template:
      apiVersion: v1
      kind: PersistentVolumeClaim
      metadata:
        name: koku-metrics-operator-data
      spec:
        accessModes:
          - ReadWriteOnce
        resources:
          requests:
            storage: 10Gi
  ```

> If a different PVC should be utilized, a valid PVC should be specified in the KokuMetricsConfig CR as described in the following section. The PVC to be used may exist already, or the operator will attempt to create it.

Reviewer: ```suggestion
If a different PVC should be utilized, a valid PVC should be specified in the KokuMetricsConfig CR as described in the appropriate section below. The PVC to be used may exist already, or the operator will attempt to create it.
```

To use the default specification, the follow assumptions must be met:
1. A default StorageClass is defined.
2. Dynamic provisioning for that default StorageClass is enabled.

> If these assumptions are not met, the operator will not deploy correctly. In these cases, storage must be manually configured. For manual configuration, a valid PVC should be supplied in the `volume_claim_template` spec of the KokuMetricsConfig CR.

Reviewer: ```suggestion
If these assumptions are not met, the operator will not deploy correctly. In these cases, storage must be manually configured. After configuring storage, a valid PVC template should be supplied in the `volume_claim_template` spec of the KokuMetricsConfig CR.
```
